PURL-TYPES.rst

```diff
@@ -397,6 +397,59 @@ nuget
 
       pkg:nuget/EnterpriseLibrary.Common@6.0.1304
 
+p2
+----
+``p2`` for Eclipse p2 units:
+
+- A p2 unit represents a unit of software that can be consumed using the Eclipse
+  Provisioning Platform. A unit can represent various types of software 
+  artifacts, such as Eclipse features, plug-ins or other bundles.
+
+- A p2 unit is uniquely identified by its id and version. A namespace is
+  therefore not necessary.
+
+- The name corresponds to the unique identifier of a given p2 unit. This id
+  uniquely identifies a unit within the Eclipse Provisioning Platform. 
+  Examples::
+
+      org.eclipse.core.resources.win32.x86_64
+      org.eclipse.rcp.sdk.id.executable.win32.win32.x86_64
+      org.eclipse.sdk.feature.group
+
+- The version corresponds to the version of the p2 unit. A p2 repository may
+  host multiple versions of the same unit. This information is therefore
+  essential when resolving the relationship between different artifacts.
+  Examples::
+
+      3.5.500.v20220812-1420
+      2.0.0.202304281106
+
+- The software artifact are accessed from a p2 repository. Given that each
+  project may host their own update site, no default repository exists. The base
+  path to this repository is provided by the ``repository_url`` qualifier key
+  and is optional.
+  Examples::
+
+       https://download.eclipse.org/releases/2023-09/202309131000
+       https://download.itemis.com/updates/releases/2.1.1
+
+- A p2 repository can host a multitude of artifacts. The type of artifact is
+  provided by the ``classifier`` qualifier key and is optional.
```

Comment:
if it's optional, a default shall be defined. I guess that `osgi.bundle` is a good one. WDYT?

Reply:
For artifacts, the type is never optional and there is no default see [IArtifactKey](https://github.com/eclipse-equinox/p2/blob/master/bundles/org.eclipse.equinox.p2.metadata/src/org/eclipse/equinox/p2/metadata/IArtifactKey.java)

Reply:
Keep in mind that my initial idea was to use units, not artifacts. So the only reason I introduced the ``classifier` was not for technical reason, but rather as a hint to immediately see what type element is described by the PURL.

With artifacts, having a classifier is now mandatory. Maven does something similar, with `jar` being the default, if no classifier is specified. In general, I like this idea, because it would reduce the length of a lot of PURLs.
But when we now put into the specification that `osgi.bundle` is assumed to be the default value, we then impose on p2 that this string must always be used to indentify bundles. That's not something I can decide...

Note that there was a similar discussion regarding Tycho recently and how it uses `p2.eclipse-plugin` as an artifical group-id for bundles that don't have proper Maven coordinates. External tools shouldn't rely on this string to always stay like this, because it's an implementation detail, rather than a formal specification. To me, this sounds like a similar situation.

Reply:
A maven `classifier` is something completely different! `jar` is the type / extension of an artifact (and the default is `jar`).
For P2 there is no default but the `classifier` can be empty (what is something different than `classifier=osgi.bundle`!) so if one needs to differentiate between _not specified_ and _empty_ what might be confusing.

Reply:
> A maven `classifier` is something completely different! `jar` is the type / extension of an artifact (and the default is `jar`). For P2 there is no default but the `classifier` can be empty (what is something different than `classifier=osgi.bundle`!) so if one needs to differentiate between _not specified_ and _empty_ what might be confusing.

That's for the clarification. Then the classifier must remain an optional hint, with no default value.

Reply:
I've updated the spec to include the feedback so far. In short, the PURL contains a:
- namespace (artifact id)
- version (artifact version)
- qualifier (classifier (optional), location (mandatory))